code/game/machinery/_machinery.dm

```diff
@@ -955,11 +955,39 @@
 	return TRUE
 
 /obj/machinery/proc/display_parts(mob/user)
-	. = list()
-	. += span_notice("It contains the following parts:")
-	for(var/obj/item/C in component_parts)
-		. += span_notice("[icon2html(C, user)] \A [C].")
-	. = jointext(., "")
+	//the initial value for an particular part/stack type. We count up from here when more instances are found
+	var/init_value = 0
```

Comment:
You shouldn't declare this all the way outside of the loop if it's only used inside the loop.

Better yet, you don't need a variable at all, directly set the index instead.

Reply:
Good point. I'll change that